src/serialization/src/main/kotlin/org/vaccineimpact/api/serialization/DataTableDeserializer.kt

```diff
@@ -66,9 +68,19 @@ open class DataTableDeserializer<out T>(
                             row: Int, column: String,
                             problems: MutableList<ErrorInfo>): Any?
     {
+        val trimmed = raw.trim()
+        if (allColumnsRequired && trimmed.isEmpty())
+        {
+            val oneIndexedRow = row + 1;
+            problems.add(ErrorInfo(
+                    "csv-missing-data:$oneIndexedRow:$column",
+                    "Unable to parse '${raw.trim()}' as ${targetType.toString().replace("kotlin.", "")} (Row $oneIndexedRow, column $column)"
+            ))
+        }
```

Comment:
Maybe pull the `allColumnsRequired` if block into a function, to keep the `deserialize` method more concise.

Reply:
e.g.
```
val trimmed = raw.trim()
checkValueIsPresentIfRequired(trimmed)
```